### ISSUE TITLE
Skip/Take Paging API

### ISSUE THREAD

**Issue by mumby0168** [IEvangelist/azure-cosmos-dotnet-repository#149]
Paging support using the `Skip()` `Take()` approach making use of offsets, which we have discussed and think will have its use cases however, in terms of cosmos DB it can cost a lot of RUs, and it's quite in-efficient. We plan to implement this API with clear documentation that states the implications of using this and where you may want to and where you should certainly not use this API.

See discussions and recent attempts at this API as well as the API which currently supports continuation tokens for a load more paging scenario #142 and this was done as part of issue #53.

The API could look something like this

```csharp
ValueTask<IPage<TItem>> PageAsync(
            Expression<Func<TItem, bool>> predicate = null,
            int pageNumber = 1,
            int pageSize = 25,
            CancellationToken cancellationToken = default);
```

**Comment by Rabosa616** [comment]
I´ve already implemented on my own, do you want me to do a PR to your Repo?

**Comment by mumby0168** [comment]
> I´ve already implemented on my own, do you want me to do a PR to your Repo?

Yeah, that sound's great 😄  be happy to see it.

**Comment by Rabosa616** [comment]
I've created the PR with the code I have in my solution

**Comment by mumby0168** [comment]
Great work @Rabosa616 PR merged, will release this shortly #153
